src/rules/noRequireImportsRule.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "require() style import is forbidden for module: ";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new NoRequireImportsWalker(sourceFile, this.getOptions()));
+    }
+}
+
+class NoRequireImportsWalker extends Lint.RuleWalker {
+
+    public visitVariableStatement(node: ts.VariableStatement) {
+        var declarations : ts.VariableDeclaration[] = node.declarationList.declarations;
+        declarations.forEach(this.processDeclaration.bind(this));
+
+        super.visitVariableStatement(node);
+    }
+
+    private processDeclaration(decl: ts.VariableDeclaration) : void {
+        // make sure the RHS is a call expression.
+        var call = <ts.CallExpression>(decl.initializer);
+        if (call && call.arguments && call.expression) {
+            var fnExpression : string = call.expression.getText(this.getSourceFile());
+            if (fnExpression === "require") {
+                var libName = call.arguments[0].getText(this.getSourceFile());
+                var errorString = Rule.FAILURE_STRING + libName;
```

Comment:
you won't need this variable once you remove "libName" from the reported error string